pkg/endpoint/policy.go

```diff
@@ -498,6 +499,230 @@ func (e *Endpoint) IngressOrEgressIsEnforced() bool {
+		l4Ports := policy.NewL4RuleContexts()
+		for _, toPort := range ig.ToPorts {
+			for _, pp := range toPort.Ports {
+				port, err := e.GetPortNumber(pp.Port)
+				if err != nil {
+					log.Debugf("Error while getting port number for %s", port)
+					continue
+				}
+				l4RuleCtx := policy.L4RuleContext{
+					Port: port,
+					// FIXME do the translation of protocols to uint8
+					// For now 0 means any
+					Proto: 0,
+				}
+				if _, ok := l4Ports[l4RuleCtx]; !ok {
+					l4Ports[l4RuleCtx] = policy.L7RuleContext{}
+				}
+			}
+			// FIXME When/How to store the ProxyPort?
+		}
+
+		// Create the policy ByNumericIdentity
+		for srcID, srcLabels := range *labelsMap {
+			for _, es := range ig.FromEndpoints {
+				// FIXME do we still need to use the policy repo to know if something is Accepted or Deny?
+				if es.Matches(srcLabels) {
+					if _, ok := ingressPolicy.ByNumericIdentity[srcID]; !ok {
+						ingressPolicy.ByNumericIdentity[srcID] = l4Ports
+					}
+				}
+			}
+		}
+
+		// Create the policy ByEntity
+		for _, entity := range ig.FromEntities {
+			ingressPolicy.ByEntity[entity] = l4Ports
+		}
+
+		// Create the policy ByCIDR
+		for _, cidr := range ig.FromCIDR {
+			ingressPolicy.ByCIDR[cidr] = l4Ports
+		}
+	}
+
+	// Create egress policy
+	for _, eg := range egress {
+		// FIXME needs validation of a singe peer on each ingress rule
+		// Accept only ToEntities or ToCIDR or ToService or ...
+
+		// Create the L4Port first
+		l4Ports := policy.NewL4RuleContexts()
+		for _, toPort := range eg.ToPorts {
+			for _, pp := range toPort.Ports {
+				port, err := e.GetPortNumber(pp.Port)
+				if err != nil {
+					log.Debugf("Error while getting port number for %s", port)
+					continue
+				}
+				l4RuleCtx := policy.L4RuleContext{
+					Port: port,
+					// FIXME do the translation of protocols to uint8
+					// For now 0 means any
+					Proto: 0,
+				}
+				if _, ok := l4Ports[l4RuleCtx]; !ok {
+					l4Ports[l4RuleCtx] = policy.L7RuleContext{}
+				}
+			}
+			// FIXME When/How to store the ProxyPort?
+		}
+
+		// FIXME We need a list of all k8s services so we can match them
+		// by labels / name
+		// The ToPorts should be enforced the policy in the Service Port and not
+		// in the endpoint port that is backed by the Service.
+
+		//for srcID, srcLabels := range *labelsMap {
+		//	for _, es := range eg.ToServices {
+		//		if es.Matches(srcLabels) {
+		//			if _, ok := ingressPolicy.ByNumericIdentity[srcID]; !ok {
+		//				egressPolicy.ByNumericIdentity[srcID] = l4Ports
+		//			}
+		//		}
+		//	}
+		//}
+
+		// Create the policy ByEntity
+		for _, entity := range eg.ToEntities {
+			egressPolicy.ByEntity[entity] = l4Ports
+		}
+
+		// Create the policy ByCIDR
+		for _, cidr := range eg.ToCIDR {
+			egressPolicy.ByCIDR[cidr] = l4Ports
+		}
+	}
+
+	// No errors so do atomic assignment
+	e.IngressPolicy = ingressPolicy
+	e.EgressPolicy = egressPolicy
+	// Q: When to bump policy revision?
+	// A: When bpf enforces it
```

Comment:
It *might* make sense to have three independent revisions: At the repository layer, at the translated policy layer here, then at the enforcement (datapath) layer?